mllib/src/main/scala/org/apache/spark/mllib/recommendation/MatrixFactorizationModel.scala

```diff
@@ -70,26 +71,28 @@ class MatrixFactorizationModel(
 
   /** Predict the rating of one user for one product. */
   def predict(user: Int, product: Int): Double = {
-    val userVector = userFeatures.lookup(user).head
-    val productVector = productFeatures.lookup(product).head
-    blas.ddot(userVector.length, userVector, 1, productVector, 1)
+    val userVector = Vectors.dense(userFeatures.lookup(user).head)
+    val productVector = Vectors.dense(productFeatures.lookup(product).head)
+    BLAS.dot(userVector, productVector)
```

Comment:
This change is not necessary. It add more wrappers (and temp objects) and it doesn't save much code.